addons/survey/static/src/question_page/question_page_list_renderer.js

```diff
@@ -96,4 +96,24 @@ export class QuestionPageListRenderer extends ListRenderer {
         }
         return super.onCellKeydownEditMode(...arguments);
     }
+
+    /**
+     * Save the survey after a question used as trigger is deleted. This allows
+     * immediate feedback on the form view as the triggers will be removed
+     * anyway on the records by the ORM.
+     *
+     * @override
+     * @param record
+     * @return {Promise<void>}
+     */
+    async onDeleteRecord(record) {
+        const triggeredRecords = this.props.list.records.filter(rec => rec.data.triggering_question_id[0] === record.data.id);
```

Comment:
Mmmh I know we discussed this together, but while testing I realized that if you click on the small bin icon by mistake you're done, it's gone forever.

I was thinking maybe a confirmation modal but I don't think Luc will like that ^^